Lib/imaplib.py

```diff
@@ -1020,6 +1020,8 @@ def _command_complete(self, name, tag):
             self._check_bye()
         try:
             typ, data = self._get_tagged_response(tag, expect_bye=logout)
+            if name == "NOOP":
+                return typ, data
```

Comment:
I don't think this is needed, if there are no untagged response from the server, otherwise, the command error is expected.